docs/src/main/asciidoc/property-source-config.adoc

```diff
@@ -415,15 +415,32 @@ NOTE:  If you use `spring.cloud.kubernetes.config.paths` or `spring.cloud.kubern
 functionality will not work.  You will need to make a `POST` request to the `/actuator/refresh` endpoint or
 restart/redeploy the application.
 
+[#config-map-fail-fast]
+In some cases, your application may be unable to load some of your `ConfigMaps` using the Kubernetes API.
+If you want your application to fail the start-up process in such cases, you can set the
+`spring.cloud.kubernetes.config.fail-fast=true` to make the application start-up fail with an Exception.
+
+[#config-map-retry]
+You can also make your application retry loading `ConfigMap` property sources on a failure. First, you need to
+set `spring.cloud.kubernetes.config.fail-fast=true`. Then you need to add `spring-retry` 
```

Comment:
it's probably obvious to you, but can I ask why `spring-retry` and `spring-boot-starter-aop` are not on the classpath and need to be added?

Reply:
Its a pattern we have used in the past with Spring Retry, ultimately we only enable retry functionality if the user specifically wants it, so we require them to include Spring Retry and aop on their app's classpath.

Reply:
so, declare them `optional` and then use `@ConditionalOnClass({ Retryable.class, Aspect.class, AopAutoConfiguration.class })`, right?

Reply:
@wind57 yes exactly. Also the Spring Cloud Config Client works in the same way :)

Reply:
thank you both, an interesting pattern.

Reply:
Ryan can I please ask for this not to be merged for a few days? I think I see a breaking change that I would like to have a look a closer look at. please.

Reply:
Sure no rush, curious what the change that you think breaks something?

Reply:
I will add my concern as soon as I reach that point in my review Ryan. It's a fabulous PR, so I will take a bit of time on it.

Reply:
@ryanjbaxter no concerns and I am good to merge, once some minor things change.